plugins/discourse-calendar/app/models/discourse_post_event/event.rb

```diff
@@ -110,21 +110,31 @@ def expired?
     def starts_at
       return nil if recurring? && recurrence_until.present? && recurrence_until < Time.current
 
-      from_event_dates =
-        event_dates.pending.order(:starts_at).last&.starts_at ||
-          event_dates.order(:updated_at, :id).last&.starts_at
+      date =
+        if association(:event_dates).loaded?
+          pending = event_dates.select { |d| d.finished_at.nil? }
```

Comment:
Edge case, but could deleted events creep in with this change?

The `event_dates.pending` previously used is:
https://github.com/discourse/discourse/blob/0466f51081c6e95b639d70cf64f74b82e78210fc/plugins/discourse-calendar/app/models/discourse_post_event/event_date.rb#L8-L13

Reply:
yes that's a good point but I think we do this in event finder directly. THis is what I did now and I added a spec to ensure index is not returning deleted events